src/jqLite.js

```diff
@@ -391,16 +401,18 @@ function jqLiteData(element, key, value) {
     var data = expandoStore && expandoStore.data;
 
     if (isSimpleSetter) { // data('key', value)
-      data[key] = value;
+      data[kebabToCamel(key)] = value;
     } else {
       if (massGetter) {  // data()
         return data;
       } else {
         if (isSimpleGetter) { // data('key')
           // don't force creation of expandoStore if it doesn't exist yet
-          return data && data[key];
+          return data && data[kebabToCamel(key)];
         } else { // mass-setter: data({key1: val1, key2: val2})
-          extend(data, key);
+          for (prop in key) {
```

Comment:
This is different than `extend`:

1. `extend` only copies own properties.
2. `extend` ignores `$$hashKey`.

If the chenges are intentional (e.g. to align with jQuery), we should clearly document them in the breaking change notice.